### PR TITLE
python3Packages.testpath: fix build reproducibility

### LINKED ISSUE #223583: python3Packages.testpath: embeds .exe executables
Building this package twice does not produce the bit-by-bit identical result each time, making it harder to detect CI breaches. You can read more about this at https://reproducible-builds.org/ .

Fixing bit-by-bit reproducibility also has additional advantages, such as avoiding hard-to-reproduce bugs, making content-addressed storage more effective and reducing rebuilds in such systems.

### Steps To Reproduce

```
nix-build '<nixpkgs>' -A python3Packages.testpath --check --keep-failed
```

You can use `diffoscope` to analyze the differences in the output of the two builds.

To view the build log of the build that produced the artifact in the binary cache:

```
nix-store --read-log $(nix-instantiate '<nixpkgs>' -A python3Packages.testpath)
```

### Additional context

https://github.com/jupyter/testpath/blob/master/testpath/commands.py#L128 suggests these executables are only used on Windows, so perhaps we should just strip them out of the package when on unix architectures?

/cc @austinbutler @costrouc

### PR DESCRIPTION
**Pull request by baloo**
###### Description of changes

Fixes #223583

This strips the exe files when testpath is used on non-windows platforms.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by raboof — APPROVED** [review]
tested the offending file is indeed gone, `stdenv.targetPlatform.isWindows` seems like the right conditional.

**Comment by raboof** [comment]
`nix-build --arg pkgs "import ./. {}" ./lib/tests/release.nix` is fine locally, ofborg was probably just having a bad day.

File: pkgs/development/python-modules/testpath/default.nix [hunk 1]
```diff
@@ -24,6 +24,12 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
+  # exe are only required when testpath is used on windows
+  # https://github.com/jupyter/testpath/blob/de8ca59539eb23b9781e55848b7d2646c8c61df9/testpath/commands.py#L128
+  preBuild = lib.optional (!stdenv.targetPlatform.isWindows) ''
```
Comment (Artturin) [inline pkgs/development/python-modules/testpath/default.nix:29]: this should be lib.optionalString

Comment (baloo) [reply pkgs/development/python-modules/testpath/default.nix:29]: oops! Thank you! https://github.com/NixOS/nixpkgs/pull/224705

**Pull request merged**